operator/src/main/resources/scripts/startServer.sh

```diff
@@ -188,12 +204,105 @@ checkEnv \
+      trace "'${DOMAIN_HOME}/servers/${SERVER_NAME}/data' does NOT exist as a directory"
+
+      # Create the server's directory in $DOMAIN_HOME/servers
+      if [ ! -d ${DOMAIN_HOME}/servers/${SERVER_NAME} ]; then
+        trace "Creating directory '${DOMAIN_HOME}/servers/${SERVER_NAME}'"
+        createFolder ${DOMAIN_HOME}/servers/${SERVER_NAME}
+      else
+        trace "'${DOMAIN_HOME}/servers/${SERVER_NAME}' already exists as a directory"
+      fi
+
+      # If server's 'data' directory is not already a symbolic link than create the symbolic link to
+      # $DATA_HOME/${SERVER_NAME}/data
+      if [ ! -L ${DOMAIN_HOME}/servers/${SERVER_NAME}/data ]; then
+        createSymbolicLink ${DATA_HOME}/${SERVER_NAME}/data ${DOMAIN_HOME}/servers/${SERVER_NAME}/data
+      else
+        trace "'${DOMAIN_HOME}/servers/${SERVER_NAME}/data' is already a symbolic link"
+      fi
+    else
+      trace "'${DOMAIN_HOME}/servers/${SERVER_NAME}/data' exists as a directory"
+
+      # server's default 'data' directory (${DOMAIN_HOME}/servers/${SERVER_NAME}/data) exists so first verify it's
+      # not a symbolic link.  If it's already a symbolic link than there is nothing to do.
+      if [ -L ${DOMAIN_HOME}/servers/${SERVER_NAME}/data ]; then
+        trace "'${DOMAIN_HOME}/servers/${SERVER_NAME}/data' is already a symbolic link"
+      else
+        # Server's default 'data' directory (${DOMAIN_HOME}/servers/${SERVER_NAME}/data) exists and is not
+        # a symbolic link so must be a directory.
+
+        # count number of files found under directory ${DOMAIN_HOME}/servers/${SERVER_NAME}/data
+        countFilesInDir ${DOMAIN_HOME}/servers/${SERVER_NAME}/data
+        fileCountServerDomainHomeDir=$?
+
+        # count number of files found under directory ${DATA_HOME}/${SERVER_NAME}/data
+        countFilesInDir ${DATA_HOME}/${SERVER_NAME}/data
+        fileCountServerDataDir=$?
+
+        # Use file counts to determine whether or not we can create a symbolic link to centralize
+        # data directory in specified ${DATA_HOME}/${SERVER_NAME}/data directory.
+        if [ ${fileCountServerDataDir} -eq 0 ]; then
+          if [ ${fileCountServerDomainHomeDir} -ne 0 ]; then
+            cp -rf ${DOMAIN_HOME}/servers/${SERVER_NAME}/data ${DATA_HOME}/${SERVER_NAME}
+            trace "Recursively copied directory/files from '${DOMAIN_HOME}/servers/${SERVER_NAME}/data' to '${DATA_HOME}/${SERVER_NAME}' directory"
+          else
+            trace "'${DOMAIN_HOME}/servers/${SERVER_NAME}/data' directory is empty"
+          fi
+
+          # forcefully delete the server's data directory so we can create symbolic link
+          frm -rf ${DOMAIN_HOME}/servers/${SERVER_NAME}/data
+          trace "Deleted directory '${DOMAIN_HOME}/servers/${SERVER_NAME}/data'"
+
+          # Create the symbolic link from server's data directory to $DATA_HOME
+          createSymbolicLink ${DATA_HOME}/${SERVER_NAME}/data ${DOMAIN_HOME}/servers/${SERVER_NAME}/data
```

Comment:
If this command fails, should the script report an error and exit?   The createSymbolicLink function itself succeeds even if the ln command it has fails.